cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java

```diff
@@ -0,0 +1,190 @@
+        when(subscription.subscriptionId()).thenReturn("dummySubscriptionId");
+        when(azureUtils.generateDesNameByNameAndId(any(), any())).thenReturn("dummyEnvName-DES-uniqueId");
+        when(azureClientService.getClient(any())).thenReturn(azureClient);
+        when(azureClient.getCurrentSubscription()).thenReturn(subscription);
+        when(azureClient.getDiskEncryptionSet(any(), any())).thenReturn(des);
+
+        CreatedDiskEncryptionSet createdDes = underTest.createDiskEncryptionSet(requestedSet);
+        assertEquals(createdDes.getDiskEncryptionSetLocation(), "dummyRegion");
+        assertEquals(createdDes.getDiskEncryptionSetResourceGroup(), "dummyResourceGroup");
+    }
+
+    @Test
+    public void testCreateDiskEncryptionSetShouldreturnNewlyCreatedDiskEncryptionSetIfNotAlreadyExists() {
+        DiskEncryptionSetCreationRequest requestedSet = new DiskEncryptionSetCreationRequest.Builder()
+                .withId("uniqueId")
+                .withCloudCredential(new CloudCredential())
+                .withRegion(Region.region("dummyRegion"))
+                .withEnvironmentName("dummyEnvName")
+                .withEnvironmentId(1L)
+                .withSingleResourceGroup(true)
+                .withResourceGroupName("dummyResourceGroup")
+                .withTags(new HashMap<>())
+                .withEncryptionKeyUrl("https://dummyVaultName.vault.azure.net/keys/dummyKeyName/dummyKeyVersion")
+                .build();
+        DiskEncryptionSetInner des = (DiskEncryptionSetInner) new DiskEncryptionSetInner()
+                .withEncryptionType(DiskEncryptionSetType.ENCRYPTION_AT_REST_WITH_CUSTOMER_KEY)
+                .withActiveKey(new KeyVaultAndKeyReference()
+                        .withKeyUrl("https://dummyVaultName.vault.azure.net/keys/dummyKeyName/dummyKeyVersion")
+                        .withSourceVault(new SourceVault()
+                                .withId("/subscriptions/dummySubs/resourceGroups/dummyResourceGroup/providers/Microsoft.KeyVault/vaults/dummyVaultName")))
+                .withIdentity(new EncryptionSetIdentity().withType(DiskEncryptionSetIdentityType.SYSTEM_ASSIGNED))
+                .withLocation("dummyRegion")
+                .withTags(new HashMap<>());
+        Subscription subscription = mock(Subscription.class);
+        when(persistenceNotifier.notifyAllocation(any(), any())).thenReturn(new ResourcePersisted());
+        when(subscription.subscriptionId()).thenReturn("dummySubscriptionId");
+        when(azureUtils.generateDesNameByNameAndId(any(), any())).thenReturn("dummyEnvName-DES-uniqueId");
+        when(azureClientService.getClient(any())).thenReturn(azureClient);
+        when(azureClient.getCurrentSubscription()).thenReturn(subscription);
+        when(azureClient.getDiskEncryptionSet(any(), any())).thenReturn(null);
+        when(azureClient.createDiskEncryptionSet(any(), any(), any(), any(), any(), any())).thenReturn(des);
```

Comment:
I think it would be nice to add the class like any(String.class) just to make sure that if the parameters are changing we still double checking the unit tests